app-modules/campaign/src/Filament/Resources/CampaignResource/Pages/CreateCampaign.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Assist\Campaign\Filament\Resources\CampaignResource\Pages;
+
+use App\Models\User;
+use Filament\Forms\Components\Select;
+use Filament\Forms\Components\Section;
+use Filament\Forms\Components\Textarea;
+use Illuminate\Database\Eloquent\Model;
+use Filament\Forms\Components\TextInput;
+use Filament\Forms\Components\Wizard\Step;
+use Filament\Resources\Pages\CreateRecord;
+use Assist\Campaign\Enums\CampaignActionType;
+use Filament\Forms\Components\DateTimePicker;
+use Assist\Campaign\Actions\CreateActionsForCampaign;
+use Assist\Engagement\Enums\EngagementDeliveryMethod;
+use Assist\Campaign\Filament\Resources\CampaignResource;
+use Filament\Resources\Pages\CreateRecord\Concerns\HasWizard;
+use Assist\Campaign\DataTransferObjects\CampaignActionCreationData;
+
+class CreateCampaign extends CreateRecord
+{
+    use HasWizard;
+
+    protected static string $resource = CampaignResource::class;
+
+    protected function getSteps(): array
+    {
+        /** @var User $user */
+        $user = auth()->user();
+
+        return [
+            Step::make('Setup your campaign')
+                ->schema([
+                    TextInput::make('name')
+                        ->required(),
+                    Select::make('caseload_id')
+                        ->label('Caseload')
+                        ->translateLabel()
+                        ->options($user->caseloads()->pluck('name', 'id'))
+                        ->searchable()
+                        ->required(),
+                ]),
+            Step::make('Create your actions')
+                ->schema([
+                    Select::make('actions')
+                        ->options(CampaignActionType::class)
+                        ->multiple()
+                        ->selectablePlaceholder(false)
+                        ->required(),
+                ]),
+            Step::make('Define your campaign data')
+                ->schema([
+                    ...$this->actionSections(),
+                ]),
```

Comment:
It's totally fine if this is planned to be done in a different PR/Ticket.

But I wonder if it may make more sense for the Action steps here to be combined into a single step with a Builder.

That way, individual Actions can be created and their Data configured. And you would only edit the Action Data of an Action type you added.

Reply:
Good call - I think that makes a ton of sense and opens up some flexibility. I will add it in this PR.